### ISSUE TITLE
Latest version 5.4.2 crashes on Android - HeadlessInAppWebView.dispose

### ISSUE THREAD

**Issue by jeromeDms** [pichillilorenzo/flutter_inappwebview#1155]
- [x] I have read the [Get Started - Installation](https://inappwebview.dev/docs/get-started/installation/) section
- [x] I have read and done the [Get Started - Setup Android](https://inappwebview.dev/docs/get-started/setup-android/) section
- [x] I have read and done the [Get Started - Setup iOS](https://inappwebview.dev/docs/get-started/setup-ios/) section
- [x] I have already searched for the same problem

## Environment

| Technology           | Version       |
| -------------------- | ------------- |
| Flutter version      |   version 2.13.0-0.2.pre    |
| Plugin version       |    5.4.2           |
| Android version      |    9           |

Device information: Huawey P20 Lite physical device

## Description

HeadlessInAppWebView crashes on dispose.
It seems the latest version changes some things in the dispose method, as mentioned in your logs.

## Code

mainApp.dart

```
SingletonServer Server = SingletonServer();

class _MyHomePageState extends State<MyHomePage> with WidgetsBindingObserver {

  @override
  void dispose() {
    WidgetsBinding.instance.removeObserver(this);
    Server.disposeServerClass();
    super.dispose();
  }
}
```

SingletonServer.dart :

```
class SingletonServer {
  static final SingletonServer _singleton = SingletonServer._internal();

  factory SingletonServer() {
    return _singleton;
  }

  SingletonServer._internal();
  late FreeWebView freeWebView;
  .......

  freeWebView = FreeWebView();

void disposeServerClass() {
    freeWebView.headlessWebView?.dispose();
  }
```

FreeWebView.dart

```
class FreeWebView {
  HeadlessInAppWebView? headlessWebView;
  InAppWebViewController? webViewController;
  .....

  headlessWebView = HeadlessInAppWebView(
      initialOptions: InAppWebViewGroupOptions(
          crossPlatform: InAppWebViewOptions(
              javaScriptEnabled: true,
          ),
    ....

    // Run the webview
    await headlessWebView?.run();
    return completer.future;
}
```

## Stacktrace/Logcat 

```
E/AndroidRuntime(29671): FATAL EXCEPTION: main
E/AndroidRuntime(29671): Process: com.xxx.yyyy, PID: 29671
E/AndroidRuntime(29671): java.lang.RuntimeException: Unable to destroy activity {com.xxx.yyyy.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.View android.app.Activity.findViewById(int)' on a null object reference
E/AndroidRuntime(29671): 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5149)
E/AndroidRuntime(29671): 	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5168)
E/AndroidRuntime(29671): 	at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:39)
E/AndroidRuntime(29671): 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
E/AndroidRuntime(29671): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
E/AndroidRuntime(29671): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2185)
E/AndroidRuntime(29671): 	at android.os.Handler.dispatchMessage(Handler.java:112)
E/AndroidRuntime(29671): 	at android.os.Looper.loop(Looper.java:216)
E/AndroidRuntime(29671): 	at android.app.ActivityThread.main(ActivityThread.java:7593)
E/AndroidRuntime(29671): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(29671): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
E/AndroidRuntime(29671): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987)
E/AndroidRuntime(29671): Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.View android.app.Activity.findViewById(int)' on a null object reference
E/AndroidRuntime(29671): 	at com.pichillilorenzo.flutter_inappwebview.headless_in_app_webview.HeadlessInAppWebView.dispose(Unknown Source:22)
E/AndroidRuntime(29671): 	at com.pichillilorenzo.flutter_inappwebview.headless_in_app_webview.HeadlessInAppWebViewManager.dispose(Unknown Source:28)
E/AndroidRuntime(29671): 	at com.pichillilorenzo.flutter_inappwebview.InAppWebViewFlutterPlugin.onDetachedFromEngine(Unknown Source:23)
E/AndroidRuntime(29671): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.remove(Unknown Source:121)
E/AndroidRuntime(29671): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.remove(Unknown Source:16)
E/AndroidRuntime(29671): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.removeAll(Unknown Source:11)
E/AndroidRuntime(29671): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.destroy(Unknown Source:10)
E/AndroidRuntime(29671): 	at io.flutter.embedding.engine.FlutterEngine.destroy(Unknown Source:31)
E/AndroidRuntime(29671): 	at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onDetach(Unknown Source:98)
E/AndroidRuntime(29671): 	at io.flutter.embedding.android.FlutterFragment.onDetach(Unknown Source:7)
E/AndroidRuntime(29671): 	at androidx.fragment.app.Fragment.performDetach(Unknown Source:6)
E/AndroidRuntime(29671): 	at androidx.fragment.app.u.i(Unknown Source:33)
E/AndroidRuntime(29671): 	at androidx.fragment.app.u.m(Unknown Source:235)
E/AndroidRuntime(29671): 	at androidx.fragment.app.c0$d.c(Unknown Source:5)
E/AndroidRuntime(29671): 	at androidx.fragment.app.c0$e.b(Unknown Source:18)
E/AndroidRuntime(29671): 	at androidx.fragment.app.c0.j(Unknown Source:215)
E/AndroidRuntime(29671): 	at androidx.fragment.app.n.T(Unknown Source:36)
E/AndroidRuntime(29671): 	at androidx.fragment.app.n.F(Unknown Source:10)
E/AndroidRuntime(29671): 	at androidx.fragment.app.i.h(Unknown Source:4)
E/AndroidRuntime(29671): 	at androidx.fragment.app.e.onDestroy(Unknown Source:5)
E/AndroidRuntime(29671): 	at android.app.Activity.performDestroy(Activity.java:7737)
E/AndroidRuntime(29671): 	at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1327)
E/AndroidRuntime(29671): 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5134)
E/AndroidRuntime(29671): 	... 11 more
```

**Comment by isyshuai** [comment]
这也发生在我身上😢

**Comment by pichillilorenzo** [comment]
I’m working on it

**Comment by pichillilorenzo** [comment]
Released new version `5.4.2+1` with the fix

**Comment by jeromeDms** [comment]
^5.4.2+1 seems to solve the issue.
But 5.4.3 has the same issue.
Normal ?
